docker/nginx/nginx.conf

```diff
@@ -0,0 +1,94 @@
+    }
+
+    fastcgi_cache_path  /etc/nginx/cache levels=1:2 keys_zone=web_cache:1m max_size=256m inactive=60m use_temp_path=off;
+    fastcgi_cache_key   "$scheme$request_method$host$request_uri";
+
+    server {
+        listen                  9200;
+        listen                  [::]:9200;
+        server_name             gewis.nl;
+        charset                 utf-8;
+        server_tokens           off;
+        auth_basic              ${NGINX_REQUIRE_AUTH};
+        auth_basic_user_file    /etc/nginx/.htpasswd;
+
+        root /code/public;
+        index index.php index.html index.htm;
+
+        location ~ ^/data/(images|javascript|js|css|flash|media|static)/  {
+            gzip_static                 on;
+            etag                        on;
+            add_header                  Cache-Control                       "private max-age=0";
+            add_header                  X-XSS-Protection                    $x_css_protection;
+            add_header                  X-Content-Type-Options              $x_content_type_options;
+            add_header                  Content-Security-Policy             $content_security_policy;
+        }
+
+        location ~ ^/(images|javascript|js|css|flash|media|static)/  {
+            gzip_static                 on;
+            etag                        on;
+            add_header                  Cache-Control                       "public max-age=0";
+            add_header                  X-XSS-Protection                    $x_css_protection;
+            add_header                  X-Content-Type-Options              $x_content_type_options;
+            add_header                  Content-Security-Policy             $content_security_policy;
+        }
+
+        location / {
+            try_files   $uri    $uri/   /index.php$is_args$args;
+        }
+
+        location ~ \.php$ {
+            # access_log /etc/nginx/logs/scripts.log scripts; # This line is useful for debugging routing errors between nginx and web
+            fastcgi_pass                web:9000;
+            fastcgi_hide_header         Expires;
+            fastcgi_hide_header         Pragma;
+            fastcgi_hide_header         Cache-Control;
+            add_header                  Cache-Control                       "private no-cache";
+            fastcgi_cache               web_cache;
+            fastcgi_cache_use_stale     error timeout http_500 http_503;
+            add_header                  X-Cache                             $upstream_cache_status;
+            add_header                  X-XSS-Protection                    $x_css_protection;
+            add_header                  X-Content-Type-Options              $x_content_type_options;
+            add_header                  Content-Security-Policy             $content_security_policy;
+        }
```

Comment:
This is related to the HTTP security headers. A good start in comparison to what we have now. We will have to manually make the filters stricter over time. There are two things I want to point out:

- `X-XSS-Protection` is functionally dead ([most major browsers have removed their XSS filters](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-XSS-Protection)) and should be removed.
- `Referrer-Policy` is missing, although we have to make sure this does not interfere with any of things that use our referer.

I am not saying we should change this now, but it is something we should keep in mind.

There is also the new `Permissions-Policy`, however, it is not widely supported and should therefore not be used. This also applies to `CORP`, `COOP`, and `COEP`.

Reply:
I agree we can indeed remove X-XSS protection if it is no longer supported.
What would you propose to do with the other policies? Feel free to make changes accordingly.

Reply:
> What would you propose to do with the other policies? Feel free to make changes accordingly.

The latter four we should not yet implement as they have limited support. `Referrer-Policy` is something we should look into at a later point (when we aren't making any major changes). Regarding the Content Security Policy, I would like to have it also report errors, but this requires additional infrastructure if we do it ourselves or an external company (e.g., Report-Uri).

So, apart from removing the `X-XSS-Protection` header I will not change anything else.

Reply:
I agree that reporting could be useful indeed. But because of the required effort, I didn't do it for now.